ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/ClasspathComputer.java

```diff
@@ -267,17 +267,23 @@ private static void addLibraryEntry(IPluginLibrary library, Map<String, IPath> s
 
 		IPackageFragmentRoot root = context.javaProject.getPackageFragmentRoot(jarFile);
 		if (root.exists() && root.getKind() == IPackageFragmentRoot.K_BINARY) {
-			IClasspathEntry oldEntry = root.getRawClasspathEntry();
-			// If we have the same binary root but new or different source, we
-			// should recreate the entry. That is when the source attachment:
-			// - is not defined: the default could be available now, or
-			// - is overridden with a different value.
-			if ((sourceAttachment == null && oldEntry.getSourceAttachmentPath() != null)
-					|| (sourceAttachment != null && sourceAttachment.equals(oldEntry.getSourceAttachmentPath()))) {
-				context.reloaded.add(oldEntry);
-				return;
+			try {
```

Comment:
should we probabbly even move the try outside the if check `getKind` is documented to also throw `JavaModelException` 

> if this element does not exist or if an
> exception occurs while accessing its corresponding resource

Reply:
i am all in for that